### PR TITLE
Move querystring parsing to the router

### PR DESCRIPTION
**Pull request by ivan-tymoshenko**
#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

Moved querystring parsing to the find-my-way router, where it parsed more efficiently. https://github.com/delvedor/find-my-way/pull/284

### CODE REVIEW

File: fastify.js [hunk 1]
```diff
@@ -131,7 +129,6 @@ function fastify (options) {
   options.logger = logger
   options.genReqId = genReqId
   options.requestIdHeader = requestIdHeader
-  options.querystringParser = querystringParser
```
Comment (climba03003) [inline fastify.js:134]: It need to be deprecate in one release cycle instead of remove it directly.

**Review by climba03003 — DISMISSED** [review]

File: fastify.js (hunk 1 above)
Comment (mcollina) [reply fastify.js:134]: There is no change in functionality with this PR.

Comment (climba03003) [reply fastify.js:134]: I may read the code wrongly, it is still passing the same option.
But, it pass to another place.

**Review by climba03003 — APPROVED** [review]
LGTM, once `find-my-way` released it.

File: lib/route.js [hunk 2]
```diff
@@ -322,7 +320,7 @@ function buildRouting (options) {
   }
 
   // HTTP request entry point, the routing has already been executed
-  function routeHandler (req, res, params, context) {
+  function routeHandler (req, res, params, context, query) {
```
Comment (jsumners) [inline lib/route.js:323]: Do we have a test that shows using `qs` over `querystring`?

Comment (ivan-tymoshenko) [reply lib/route.js:323]: There is a [custom-querystring-parser.test.js](https://github.com/fastify/fastify/blob/main/test/custom-querystring-parser.test.js) that tests the custom parser option. You can use any querystring parser that you want. I don't see the need to test particular implementations.

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**